### ISSUE TITLE
FolderOpenDialog SetOpenText and SetDismissText are ineffective

### ISSUE THREAD

**Issue by BabySid** [fyne-io/fyne#2830]
### Describe the bug:
As the title describes, when i use a folderopendialog, i cannot found any method that set the text of the OPEN button.
And, I call SetDismissText, but it does't have any effect.

### To Reproduce:
Steps to reproduce the behaviour:
- Run the code below

### Screenshots:
![图片](https://user-images.githubusercontent.com/6658943/157486632-8d279859-3eed-4e5a-b0c8-ff490c1eaa4d.png)

### Example code:
```go
func main() {
        myApp := app.NewWithID("go-fyne")

	myWindow := myApp.NewWindow("Widget")
	btn := widget.NewButton("Go", func() {
		fo := dialog.NewFolderOpen(func(uri fyne.ListableURI, err error) {
			if uri != nil {
				fmt.Println(uri.Path())
			}
		}, myWindow)
		fo.SetDismissText("Exit")
		fo.Show()
	})

	content := container.NewVBox(
		widget.NewEntry(),
	)

	myWindow.Resize(fyne.NewSize(800, 600))
	myWindow.SetContent(container.NewBorder(btn, nil, nil, nil, content))
	myWindow.CenterOnScreen()
	myWindow.ShowAndRun()
}
```

### Device (please complete the following information):
 - **OS:** Windows10 
 - **Version:** 19044
 - **Go version:** go1.17.5
 - **Fyne version:** 2.1.3

**Comment by andydotxyz** [comment]
Thanks for this.
We can resolve the `SetDismissText` not functioning in the next point release, but adding the setting of save text will have to wait a little longer as we don't add API (which this requires) in bug-fix releases.
I will open a PR with `SetConfirmText` as this matches other dialogs, and works for save/open equally.

**Comment by andydotxyz** [comment]
The `SetDismissText` oversight is fixed pending v2.1.4

**Comment by andydotxyz** [comment]
The bug fix is released, the new API for SetConfirmText will be in v2.2.0
